app/src/main/java/com/example/event_finder_capstone/activities/EventDetailsActivity.java

```diff
@@ -83,6 +90,19 @@ public boolean onDoubleTap(MotionEvent e) {
         getWindow().getDecorView().setOnTouchListener((view, event) -> detector.onTouchEvent(event));
     }
 
+    private int checkIfEventAlreadyBookmarked(String eventId) throws com.parse.ParseException {
+        ParseQuery<Bookmark> query = ParseQuery.getQuery(Bookmark.class);
+        query.whereEqualTo(Bookmark.KEY_BOOKMARKED_EVENT_ID, eventId);
+        query.whereEqualTo(Bookmark.KEY_USER, ParseUser.getCurrentUser());
+        query.findInBackground((bookmarks, e) -> {
+            if (e != null) {
+                Toast.makeText(EventDetailsActivity.this, "Failed to check if already bookmarked", Toast.LENGTH_LONG).show();
+                return;
+            }
+        });
+        return query.count();
```

Comment:
🔍 Nit:  You never know what users do, but I suppose it's theoretically possible if a user double-taps fast enough they could bookmark twice.  To be safe, I'd probably check if `query.count() >= 1`.

Additionally, it might be unclear to other engineers how to use this method, which returns an integer.  I'd personally return a boolean here instead of the count.